### PR TITLE
add missing `Send` impl for `AHeap`

### PR DESCRIPTION
**Pull request by hawkw**
Currently, any `Future` that awaits one of the `AHeap` `allocate`,
`allocate_arc`, `allocate_array_with`, or `allocate_fixed_vec` methods
is `!Send`. This is because these methods return a `Future` which
contains a `&self` receiver, and the `AHeap` type itself is `!Send`.
This is sad and not very great.

This branch adds a missing `Send` impl to `AHeap`. This should be safe
for the same reason that the existing `unsafe impl Sync for AHeap` is
safe.

### CODE REVIEW

**Review by jamesmunns — APPROVED** [review]
So, I think this is correct today, I sorta worry that AHeap isn't *really* send - moving it (anywhere!) would invalidate everything allocated with the given heap, due to the ptr back to the AHeap.

That being said, there is no (safe) way to create a stack allocated AHeap, it can only be created with `bootstrap`, which already assumes the data it is given has `'static` lifetime, and I don't think there is any public way to get the `&mut Heap` (there's a private method on the HeapGuard), which could allow you to `core::mem::swap` (we should probably turn that interface into Pin just in case, I think).

Basically, I don't know how to express "whew don't move it", which is normally what I think of with Send. But lets go for this, and maybe this is just another straw in the pile of "egg Eliza into rewriting the whole allocator".
